src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNIndexCache.java

```diff
@@ -236,4 +236,8 @@ public void onFileDeleted(Path indexFilePath) {
             getInstance().cache.invalidate(indexFilePath.toString());
         }
     };
+
+    private String[] getQueryParams(String indexName) {
+        return new String[] {"efSearch=" + KNNSettings.getEfSearchParam(indexName)};
```

Comment:
Could you explain more why there is only 1 query parameter being passed in? Why are M and efConstruction not gotten?

Reply:
```efSearch``` is required only during searches.  Other algorithm params are needed for indexing.